.github/workflows/master.yml

```diff
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 Canonical Ltd.
+
+name: Master workflow
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  build-dbtestapp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build
+        run: |
+          cd mongoapi/dbtestapp/
+          go build
+
+  docker-build-dbtestapp:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build and push Docker image
+        run: |
+          cd mongoapi/dbtestapp
+          make docker-build
+          make docker-push
```

Comment:
Should we do this for all of the repos? That is, add `make docker-push` to the actions that are doing `make docker-build`?